src/library_webgpu.js

```diff
@@ -94,11 +94,12 @@ var LibraryWebGPU = {
 
       function makeManager() {
         return {
-          objects: [undefined],
+          objects: {},
+          nextId: 1,
           create: function(object, wrapper /* = {} */) {
             wrapper = wrapper || {};
 
-            var id = this.objects.length;
+            var id = this.nextId++;
```

Comment:
The array scheme from before has an issue that the length of `objects` array will unboundedly grow, even if one deleted all the objects one created. The result would be an array with first element being `undefined`, and then tens of thousands of `empty` elements, as shown by Chrome DevTools.

It is uncertain if the empty elements actually retained any memory allocations, but better to be on the safe side. (e.g. how much memory would browser allocate for `[undefined, 100000xempty, {}]`)

Reply:
Browsers have to correctly handle very sparse arrays; arrays in JavaScript are just objects (`{}`) keyed by the string representations of their indices. Non-sparse arrays are an engine optimization. Though it's possible that this particular pattern (using and then deleting every index) could cause memory issues I suppose. So this change seems fine.